### ISSUE TITLE
Let Hibernate detect the dialect to use

### ISSUE THREAD

**Issue by snicoll** [spring-projects/spring-boot#16172]
Spring Boot detects the dialect to use based on the `DataSource`. Unfortunately, the defaults in framework are quite conservative and Hibernate is doing a good job at detecting the right dialect anyway.

As suggested in https://github.com/spring-projects/spring-boot/issues/15342#issuecomment-461001649 we should stop doing that upfront and let Hibernate do its thing.

**Comment by viktoriya-kutsarova** [comment]
@snicoll If this issue is suitable for contribution, I can try to tackle it.

**Comment by snicoll** [comment]
Thanks for the offer. The issue is assigned and therefore not opened for contribution. I've assigned myself as there are quite a lot of things to consider.

**Comment by trajano** [comment]
I don't think this is working yet.  I put a breakpoint on StandardDialectResolver and I found that it is not being hit with spring-boot 2.1.7

**Comment by snicoll** [comment]
@trajano that's because this was fixed in Spring Boot 2.2. Perhaps you're looking at github on your phone and you're missing the column that provides the milestone on the right?

**Comment by trajano** [comment]
No I just noticed it now.  Since this was closed about 4 months ago I thought it would've been here by now.

Anyway for those who are still stuck on the older version in case they reach here my workaround was

```
@Component
@Primary
public class CustomHibernateJpaVendorAdapter extends HibernateJpaVendorAdapter {

    @Override
    protected Class<?> determineDatabaseDialectClass(Database database) {
        return null;
    }

}
```
